doc/sphinx/docs/python/tutorials/coding_udp_multi_objective.rst

```diff
@@ -0,0 +1,70 @@
+.. _py_tutorial_coding_udp_multi_objective:
+
+Coding a User Defined Problem with constraints multiple objectives
+------------------------------------------------------------------
+
+In this chapter we show how to code an unconstrained user defined problem (UDP) with multiple objectives.
+We assume that the mathematical formulation of problem is the following:
+
+\begin{array}{rl}
+\mbox{minimize:} & f_{1}\left(x\right) = x^{2} \\
+& f_{2}\left(x\right) = \left(x-2\right)^{2} \\
+\mbox{subject to:} & 0 <= x <= 2\\
+\end{array}
+
+which is a test function for multi-objective optimization being introduced in
+*Schaffer, J. David (1984). Some experiments in machine learning using vector
+evaluated genetic algorithms (artificial intelligence, optimization, adaptation,
+pattern recognition) (PhD). Vanderbilt University.* and illustrated
+`here <https://en.wikipedia.org/wiki/Test_functions_for_optimization#Test_functions_for_multi-objective_optimization>`_.
+
+The implementation as UDP can be realized as follows:
+
+.. doctest::
+
+    >>> import pygmo as pg
+
+    >>> class Schaffer():
+    ...     # Define objectives
+    ...     def fitness(self, x):
+    ...         f1 = x[0]**2
+    ...         f2 = (x[0]-2)**2
+    ...         return [f1, f2]
+    ...
+    ...     # Return number of objectives
+    ...     def get_nobj(self):
+    ...         return 2
+    ...
+    ...     # Return bounds of decision variables
+    ...     def get_bounds(self):
+    ...         return ([0]*1, [2]*1)
+    ...
+    ...     # Return function name
+    ...     def get_name(self):
+    ...         return "Schaffer function N.1"
+
+Note that the only difference between a mono- and multi-objective problem lies in the number of objectives.
+
+Let's now create an object from our new UDP class and pass it to a pygmo :class:`~pygmo.problem`.
+
+.. doctest::
+
+    >>> # create UDP
+    >>> prob = pg.problem(Schaffer())
+
+In the next step, the problem can be solved straightforward using the NSGA2 algorithm from the :class:`~pygmo.algorithm` class:
+
+.. doctest::
+
+    >>> # create population
+    >>> pop = pg.population(prob, size=200)
+    >>> # select algorithm
+    >>> algo = pg.algorithm(pg.nsga2(gen=40))
+    >>> # run optimization
+    >>> pop = algo.evolve(pop)
+    >>> # print results
+    >>> fits, vectors = pop.get_f(), pop.get_x()
+    >>> print(fits[:1]) #doctest: +SKIP
```

Comment:
Not sure if printing one fitness says anything on the problem being solved. Maybe here the tutorial could benefit from the extraction of the non dominated front? Or its plot?

Reply:
I have printed the NDF. See: https://github.com/esa/pagmo2/pull/191/commits/99206785bfb53f8fe36738190f69fcf57f316d57